src/common.jl

```diff
@@ -179,6 +193,33 @@ function SciMLBase.init(prob::LinearProblem, alg::SciMLLinearSolveAlgorithm,
     return cache
 end
 
+
+function SciMLBase.reinit!(cache::LinearCache;
```

Comment:
Just for my understanding here:
1.  Is it intended that the input cache is not modified? I.e. shouldn't we have something like if A !== nothing then cache.A = A and so on?
2. I think the preconditioners are still not updated if we do `cache.A = A`. Is this intended?

Reply:
1. Logic bug. Fix incoming.
2. updating preconditioners with a manual setting of `cache.A` (or worse, mutating `cache.A`). I think it will be required for users to `reinit!` if they want this functionality.

Reply:
2. xref https://github.com/SciML/LinearSolve.jl/blob/11441b1ca37ca11ec946da76722c8fb5a2277ae5/src/common.jl#L86-L98